### ISSUE TITLE
Get UV coordinates from igl::LSCM in same coordinate space as mesh?

### ISSUE THREAD

**Issue by Linchy** [libigl/libigl#493]
Hi,  I would like to pass a mesh (with seams) into the LSCM algorithm, such as this cube:

![image](https://cloud.githubusercontent.com/assets/1840551/24039377/d99a50e6-0afc-11e7-874d-5128a103b4ee.png)

and get the returned texture coordinates in the same coordinate space as the original object, like so:

![image](https://cloud.githubusercontent.com/assets/1840551/24039599/b79aac88-0afd-11e7-8957-0e4653dbe3df.png)

(note that 5px is arbitrary, the mesh could have any dimensions)

So far I have the following code (calling C++ from C# via interop):

```C++
extern "C"
{	
	__declspec(dllexport) void CalculateUVCoordinates(
		float* vertices, int vertCount, int* indices, int faceCount, float* out_UVs)
	{
		// convert vertices to matrix
		Eigen::MatrixXd V;
		V.resize(vertCount, 3);
		for (int i = 0; i < vertCount; i++)
		{
			int j = i * 3;
			V(i, 0) = vertices[j];
			V(i, 1) = vertices[j + 1];
			V(i, 2) = vertices[j + 2];
		}

		// convert faces to matrix
		Eigen::MatrixXi F;
		F.resize(faceCount, 3);
		for (int i = 0; i < faceCount; i++)
		{
			int j = i * 3;
			F(i, 0) = indices[j];
			F(i, 1) = indices[j + 1];
			F(i, 2) = indices[j + 2];
		}

		// Get boundary
		VectorXi bnd;
		igl::boundary_loop(F, bnd);
		if (bnd.size() <= 1)
			return;

		// Get two points on the boundary I think?
		VectorXi b(2, 1);
		b(0) = bnd(0);
		b(1) = bnd(round(bnd.size() / 2));

		// not sure what this does
		MatrixXd bc(2, 2);
		bc << 0, 0, 1, 0;

		// LSCM parametrization
		Eigen::MatrixXd V_uv;
		igl::lscm(V, F, b, bc, V_uv);

		if (V_uv.rows() != vertCount)
			return;
		
		// convert the UV matrix back to an array
		for (int i = 0; i < vertCount; i++)
		{
			int j = i * 2;
			out_UVs[j + 0] = V_uv(i, 0);
			out_UVs[j + 1] = V_uv(i, 1);
		}
	}
}
```

But this seems to return coordinates in the [0, 1] range. I understand it has something to do with the `bc` value, though am unsure how to change it to get what I'm after.

**Comment by jiangzhongshi** [comment]
I assume, for parameterization algorithms in libigl, `b` stands for boundary and `bc` for boundary conditions. 

So `b` consists of a vector of boundary indices you want to specify, and each corresponding row of `bc` stores the coordinate you want them to be.

In your case, just find two of the corners `b << c0,c1`and store their desired uv-positions in `bc.row(0)` and `bc.row(1)` (you may need to cut the mesh first)

**Comment by Linchy** [comment]
Hi @jiangzhongshi, I tried out what you said doing the following:

1. Found an edge on the boundary
2. Set `edgeLength` = to the distance between the first and second edge vertex
3. Passed in the edge vertex indices like so `b << edgeIndex1, edgeIndex2;`
4. Passed in `<< 0, 0, edgeLength, 0`

This seems to work on and off depending on the mesh I pass. Some produce tiny UV maps like 10x10 pixels apart, which is clearly the wrong resolution when applied to the model as the texture is scaled up something like 50x. Does my process seem right?
